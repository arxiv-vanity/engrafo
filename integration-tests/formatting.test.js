var utils = require("./utils");

test("lists render correctly", done => {
  utils.expectBodyToMatchSnapshot("formatting/lists.tex", done);
});

test("text formatting renders correctly", done => {
  utils.expectBodyToMatchSnapshot("formatting/text.tex", done);
});

test("links get wrapped in anchors", done => {
  utils.expectBodyToMatchSnapshot("formatting/links.tex", done);
});

test("vskip is ignored", done => {
  utils.expectBodyToMatchSnapshot("formatting/vskip.tex", done);
});

test("\pdfoutput=1 is ignored", done => {
  utils.expectBodyToMatchSnapshot("formatting/pdfoutput.tex", done);
});

// FIXME: needs latexml bindings. which papers is this in?
test.skip("pinforms3 TITLE, AUTHOR, AFF are supported", done => {
  utils.expectBodyToMatchSnapshot("formatting/pinforms3.tex", done);
});

// FIXME: needs latexml bindings
test.skip("ICML commands are supported", done => {
  utils.expectBodyToMatchSnapshot("formatting/icml.tex", done);
});

// FIXME: needs latexml bindings for \inst - where is this from?
test.skip("\institute command is supported", done => {
  utils.expectBodyToMatchSnapshot("formatting/institute.tex", done);
});
