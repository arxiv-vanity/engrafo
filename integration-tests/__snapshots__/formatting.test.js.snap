// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`links get wrapped in anchors 1`] = `
<body>
  <div
    class="engrafo-container"
  >
    


    <div
      class="ltx_page_content"
    >
      

      <dt-article>
        

        <div
          class="ltx_para"
          id="p1"
        >
          

          <p
            class="ltx_p"
          >
            <span>
              http link: 
              <a
                href="http://link.co.uk/to#something"
              >
                http://link.co.uk/to#something
              </a>
            </span>
          </p>
          

        </div>
        

        <div
          class="ltx_para"
          id="p2"
        >
          

          <p
            class="ltx_p"
          >
            <span>
              https link: 
              <a
                href="https://link.info?param"
              >
                https://link.info?param
              </a>
               url
            </span>
          </p>
          

        </div>
        

        <div
          class="ltx_para"
          id="p3"
        >
          

          <p
            class="ltx_p"
          >
            <span>
              link with trailing period: 
              <a
                href="https://example.com."
              >
                https://example.com.
              </a>
               (FIXME: known failure)
            </span>
          </p>
          

        </div>
        

        <div
          class="ltx_para"
          id="p4"
        >
          

          <p
            class="ltx_p"
          >
            this should not make a link: foo.bar
          </p>
          

        </div>
        

        <div
          class="ltx_para"
          id="p5"
        >
          

          <p
            class="ltx_p"
          >
            <span>
              this is a link without leading space: badlatex.
              <a
                href="http://example.com"
              >
                http://example.com
              </a>
            </span>
          </p>
          

        </div>
        

        <div
          class="ltx_para"
          id="p6"
        >
          

          <p
            class="ltx_p"
          >
            this is already a link: 
            <a
              class="ltx_ref ltx_href"
              href="http://example.com"
              title=""
            >
              example.com
            </a>
          </p>
          

        </div>
        

        <div
          class="ltx_para"
          id="p7"
        >
          

          <p
            class="ltx_p"
          >
            this is a link without http: 
            <a
              class="ltx_ref ltx_href"
              href="example.com"
              title=""
            >
              example.com
            </a>
          </p>
          

        </div>
        

      </dt-article>
      

      <dt-appendix>
        

        


        <div
          class="l-body"
        />
        

      </dt-appendix>
    </div>
    





    <div
      id="footnote-hover-boxes-container"
    />
  </div>
</body>
`;

exports[`lists render correctly 1`] = `
<body>
  <div
    class="engrafo-container"
  >
    


    <div
      class="ltx_page_content"
    >
      

      <dt-article>
        

        <div
          class="ltx_para"
          id="p1"
        >
          

          <ul
            class="ltx_itemize"
          >
            

            <li
              class="ltx_item"
              id=".i1"
            >
              
 

              <div
                class="ltx_para"
                id=".i1.p1"
              >
                

                <p
                  class="ltx_p"
                >
                  An unordered list
                </p>
                

              </div>
              

            </li>
            

            <li
              class="ltx_item"
              id=".i2"
            >
              
 

              <div
                class="ltx_para"
                id=".i2.p1"
              >
                

                <p
                  class="ltx_p"
                >
                  Second item
                </p>
                

              </div>
              

            </li>
            

          </ul>
          

        </div>
        

        <div
          class="ltx_para"
          id="p2"
        >
          

          <ol
            class="ltx_enumerate"
          >
            

            <li
              class="ltx_item"
            >
              
 

              <div
                class="ltx_para"
                id="p2.p1"
              >
                

                <p
                  class="ltx_p"
                >
                  An ordered list
                </p>
                

              </div>
              

            </li>
            

            <li
              class="ltx_item"
            >
              
 

              <div
                class="ltx_para"
                id="p2.p2"
              >
                

                <p
                  class="ltx_p"
                >
                  Second one.
                </p>
                

              </div>
              

            </li>
            

          </ol>
          

        </div>
        

        <div
          class="ltx_para"
          id="p3"
        >
          

          <ul
            class="ltx_itemize"
          >
            

            <li
              class="ltx_item"
              id=".i1a"
            >
              
 

              <div
                class="ltx_para"
                id=".i1a.p1"
              >
                

                <p
                  class="ltx_p"
                >
                  A nested unordered list
                </p>
                

                <ul
                  class="ltx_itemize"
                >
                  

                  <li
                    class="ltx_item"
                    id=".i1b"
                  >
                    
 

                    <div
                      class="ltx_para"
                      id=".i1b.p1"
                    >
                      

                      <p
                        class="ltx_p"
                      >
                        First nested item
                      </p>
                      

                    </div>
                    

                  </li>
                  

                  <li
                    class="ltx_item"
                    id=".i2a"
                  >
                    
 

                    <div
                      class="ltx_para"
                      id=".i2a.p1"
                    >
                      

                      <p
                        class="ltx_p"
                      >
                        Second nested item
                      </p>
                      

                    </div>
                    

                  </li>
                  

                </ul>
                

              </div>
              

            </li>
            

            <li
              class="ltx_item"
              id=".i2b"
            >
              
 

              <div
                class="ltx_para"
                id=".i2b.p1"
              >
                

                <p
                  class="ltx_p"
                >
                  Second item
                </p>
                

              </div>
              

            </li>
            

          </ul>
          

        </div>
        

        <section
          class="ltx_section"
          id=".i2c"
        >
          

          <h2
            class="ltx_title ltx_title_section"
          >
            

            <span
              class="ltx_tag ltx_tag_section"
            >
              <a
                href="#.i2c"
              >
                <span
                  class="ltx_ERROR undefined"
                >
                  \\thesection
                </span>
                 
              </a>
            </span>
            A nested ordered list
          </h2>
          


          <div
            class="ltx_para"
            id=".i2c.p1"
          >
            

            <ol
              class="ltx_enumerate"
            >
              

              <li
                class="ltx_item"
              >
                
 

                <div
                  class="ltx_para"
                  id=".i2c.p1.p1"
                >
                  

                  <p
                    class="ltx_p"
                  >
                    First item
                  </p>
                  

                  <ol
                    class="ltx_enumerate"
                  >
                    

                    <li
                      class="ltx_item"
                    >
                      
 

                      <div
                        class="ltx_para"
                        id=".i2c.p1.p1.p1"
                      >
                        

                        <p
                          class="ltx_p"
                        >
                          First nested item
                        </p>
                        

                      </div>
                      

                    </li>
                    

                    <li
                      class="ltx_item"
                    >
                      
 

                      <div
                        class="ltx_para"
                        id=".i2c.p1.p1.p2"
                      >
                        

                        <p
                          class="ltx_p"
                        >
                          Second nested item
                        </p>
                        

                      </div>
                      

                    </li>
                    

                  </ol>
                  

                </div>
                

              </li>
              

              <li
                class="ltx_item"
              >
                
 

                <div
                  class="ltx_para"
                  id=".i2c.p1.p2"
                >
                  

                  <p
                    class="ltx_p"
                  >
                    Second item
                  </p>
                  

                </div>
                

              </li>
              

            </ol>
            

          </div>
          

        </section>
        

      </dt-article>
      

      <dt-appendix>
        

        


        <div
          class="l-body"
        />
        

      </dt-appendix>
    </div>
    





    <div
      id="footnote-hover-boxes-container"
    />
  </div>
</body>
`;

exports[`pdfoutput=1 is ignored 1`] = `
<body>
  <div
    class="engrafo-container"
  >
    


    <div
      class="ltx_page_content"
    >
      

      <dt-article>
        

        <div
          class="ltx_para"
          id="p1"
        >
          

          <p
            class="ltx_p"
          >
            hello world
          </p>
          

        </div>
        

      </dt-article>
      

      <dt-appendix>
        

        


        <div
          class="l-body"
        />
        

      </dt-appendix>
    </div>
    





    <div
      id="footnote-hover-boxes-container"
    />
  </div>
</body>
`;

exports[`text formatting renders correctly 1`] = `
<body>
  <div
    class="engrafo-container"
  >
    


    <div
      class="ltx_page_content"
    >
      

      <dt-article>
        

        <div
          class="ltx_para"
          id="p1"
        >
          

          <p
            class="ltx_p"
          >
            This is 
            <span
              class="ltx_text ltx_font_bold"
            >
              bold
            </span>
            .
          </p>
          

        </div>
        

        <div
          class="ltx_para"
          id="p2"
        >
          

          <p
            class="ltx_p"
          >
            This is 
            <span
              class="ltx_text ltx_font_italic"
            >
              textit
            </span>
            .
          </p>
          

        </div>
        

        <div
          class="ltx_para"
          id="p3"
        >
          

          <p
            class="ltx_p"
          >
            This is 
            <em
              class="ltx_emph ltx_font_italic"
            >
              emph
            </em>
            .
          </p>
          

        </div>
        

        <div
          class="ltx_para"
          id="p4"
        >
          

          <p
            class="ltx_p"
          >
            This is 
            <span
              class="ltx_text ltx_font_typewriter"
            >
              fixed width
            </span>
            .
          </p>
          

        </div>
        

      </dt-article>
      

      <dt-appendix>
        

        


        <div
          class="l-body"
        />
        

      </dt-appendix>
    </div>
    





    <div
      id="footnote-hover-boxes-container"
    />
  </div>
</body>
`;

exports[`vskip is ignored 1`] = `
<body>
  <div
    class="engrafo-container"
  >
    


    <div
      class="ltx_page_content"
    >
      

      <dt-article>
        

        <div
          class="ltx_para"
          id="p1"
        >
          

          <p
            class="ltx_p"
          >
            before
          </p>
          

        </div>
        

        <div
          class="ltx_para"
          id="p2"
        >
          

          <p
            class="ltx_p"
          >
            after
          </p>
          

        </div>
        

      </dt-article>
      

      <dt-appendix>
        

        


        <div
          class="l-body"
        />
        

      </dt-appendix>
    </div>
    





    <div
      id="footnote-hover-boxes-container"
    />
  </div>
</body>
`;
